withdraw-by-check/REPWRITERSPECS/BANNO.CHECK.WITHDRAW.V1.POW

```diff
@@ -797,174 +915,160 @@ PROCEDURE CONFIGFILEPOSTPROCESS
+  END
+
+ IF FOUNDINDEX<=PARAMERRDISPLAYMAX THEN
+  DO
+   IF FOUND=FALSE THEN
+    DO
+     PARAMERRDISPLAYCOUNT=FOUNDINDEX
+     PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYERRCODE)=TMPERRCODE
+     PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYLINECOUNT)=1
+     PARAMERRDISPLAYLINES(FOUNDINDEX,1)=TMPDISPLAYLINE
+    END
+   ELSE IF PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYLINECOUNT)<PARAMERRDISPLAYLINESMAX THEN
+    DO
+     PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYLINECOUNT)=PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYLINECOUNT)+1
+     PARAMERRDISPLAYLINES(FOUNDINDEX,PARAMERRDISPLAY(FOUNDINDEX,ERRDISPLAYLINECOUNT))=TMPDISPLAYLINE
+    END 
   END
- IF LENGTH(TMPCHR)=0 THEN
-  TMPCHR=""
 END [PROCEDURE]
 
-PROCEDURE NTS
-[* Remove trailing spaces
-**  PASS       TMPCHR
-**  RETURN     TMPCHR
+[*
+**  --------------------------------------------------------------------
+**  Debug/Error Processing Procedures
+**  --------------------------------------------------------------------
+*]
+PROCEDURE BNOERRADD
+[* Adds an error code to the valid list of error codes
+**
+**  PASS    BNOERRCODE       Numeric error code
+**  PASS    BNOERRMESSAGE    Error description
 *]
- IF TMPCHR<>"" THEN
-  TMPCHR=SEGMENT(TMPCHR,1,LENGTH(TMPCHR))
- IF LENGTH(TMPCHR)=0 THEN
-  TMPCHR=""
 
+ IF BNOERRCOUNT<BNOERRMAX THEN
+  DO
+   BNOERRCOUNT=BNOERRCOUNT+1
+   BNOERRCODES(BNOERRCOUNT)=BNOERRCODE
+   BNOERRMESSAGES(BNOERRCOUNT)=BNOERRMESSAGE
+  END
+ BNOERRMESSAGE=""
 END [PROCEDURE]
 
-PROCEDURE CLEANUPTEXT
-[* Check character string and remove out any characters or hidden
-** characters which have an ASCII value which is not established in
-** the VALIDCHRINPUT1 string in the SETUP division . This is to avoid
-** display issues with displaying the terms or conditions. Additionally,
-** if a back-slash or double quote is encountered, add a preceding
-** backslash
-**  PASS       TMPCHR   String to test
-**  RETURN     TMPCHR   Updated string
+PROCEDURE BNOERRUPDATE
+[* Updates an error code error message or display message
+**
+**  PASS    BNOERRCODE       Numeric error code
+**  PASS    BNOERRMESSAGE    Error description
+**  PASS    BNOERRDISPLAYMSG Error display message
 *]
- LELISTINPUT=VALIDCHRINPUT1
- CALL LISTEXPAND
- FOR TMPLOOP=0 TO MAXCHRVALUE
-  DO
-   VALIDCHARS(TMPLOOP)=LELIST(TMPLOOP)
-  END
- TMPTEXT=""
 
- IF TMPCHR<>"" THEN
+ BNOERRTMPCOUNT=1
+ BNOERRFOUND=FALSE
+ WHILE BNOERRTMPCOUNT<=BNOERRCOUNT AND 
+       BNOERRFOUND=FALSE
   DO
-   FOR TMPTEXTLOOP=1 TO LENGTH(TMPCHR)
+   IF BNOERRCODES(BNOERRTMPCOUNT)=BNOERRCODE THEN
     DO
-     CHRVAL=CHRVALUE(SEGMENT(TMPCHR,TMPTEXTLOOP,TMPTEXTLOOP))
-     IF  VALIDCHARS(CHRVAL)=TRUE THEN
-      DO
-[* Check for back-slash or double-quote. If found, precede with a back-slash
-*]
-       IF (CHRVAL=92 OR CHRVAL=34) THEN
-        TMPTEXT=TMPTEXT+"\"+SEGMENT(TMPCHR,TMPTEXTLOOP,TMPTEXTLOOP)
-       ELSE
-        TMPTEXT=TMPTEXT+SEGMENT(TMPCHR,TMPTEXTLOOP,TMPTEXTLOOP)
-      END
+     BNOERRFOUND=TRUE
+     IF BNOERRMESSAGE<>"" THEN
```

Comment:
Agree.  Originally, that procedure could also update the BNOERRDISPLAYMSG.  That is why I had that because there could have been cases where BNOERRMESSAGE was not being updated.  Fixed.